src/main/java/cuchaz/enigma/analysis/BuiltinPlugin.java

```diff
@@ -0,0 +1,167 @@
+	}
+
+	// TODO make this actually work; currently the resolver does not resolve specific methods that implement bridge methods
+	private void registerBridgeNamingService(EnigmaPluginContext ctx) {
+		ctx.registerService("enigma:specialized_bridge_method_indexer", JarIndexerService.TYPE, ctx1 -> (classCache, jarIndex) -> {BuiltinPlugin.this.specializedToBridge = jarIndex.getBridgeMethodIndex()::getBridgeFromSpecialized;
+			jarIndex.getBridgeMethodIndex().getSpecializedToBridge().entrySet().forEach(System.out::println);});
+		ctx.registerService("enigma:specialized_method_name_proposer", NameProposalService.TYPE, ctx1 -> (obfEntry, remapper) -> {
+			System.out.println(obfEntry);
+			if (!(obfEntry instanceof MethodEntry))
+				return Optional.empty();
+			MethodEntry bridge = BuiltinPlugin.this.specializedToBridge.apply((MethodEntry) obfEntry);
+			if (bridge == null)
+				return Optional.empty();
+			return Optional.of(remapper.deobfuscate(remapper.getObfResolver().resolveFirstEntry((MethodEntry) obfEntry, ResolutionStrategy.RESOLVE_ROOT)).getName()); // People never name bridge methods!
+		});
+	}
+
+	private static final class EnumFieldNameFindingVisitor extends ClassVisitor {
+
+		private ClassEntry clazz;
+		private String className;
+		private final Map<Entry<?>, String> mappings;
+		private final Set<Pair<String, String>> enumFields = new HashSet<>();
+		private final List<MethodNode> classInits = new ArrayList<>();
+
+		EnumFieldNameFindingVisitor(Map<Entry<?>, String> mappings) {
+			super(Opcodes.ASM7);
+			this.mappings = mappings;
+		}
+
+		@Override
+		public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
+			super.visit(version, access, name, signature, superName, interfaces);
+			this.className = name;
+			this.clazz = new ClassEntry(name);
+			this.enumFields.clear();
+			this.classInits.clear();
+		}
+
+		@Override
+		public FieldVisitor visitField(int access, String name, String descriptor, String signature, Object value) {
+			if ((access & Opcodes.ACC_ENUM) != 0) {
+				if (!enumFields.add(new Pair<>(name, descriptor))) {
+					throw new IllegalArgumentException("Found two enum fields with the same name \"" + name + "\" and desc \"" + descriptor + "\"!");
+				}
+			}
+			return super.visitField(access, name, descriptor, signature, value);
+		}
+
+		@Override
+		public MethodVisitor visitMethod(int access, String name, String descriptor, String signature, String[] exceptions) {
+			if ("<clinit>".equals(name)) {
+				MethodNode node = new MethodNode(api, access, name, descriptor, signature, exceptions);
+				classInits.add(node);
+				return node;
+			}
+			return super.visitMethod(access, name, descriptor, signature, exceptions);
+		}
+
+		@Override
+		public void visitEnd() {
+			super.visitEnd();
+			try {
+				collectResults();
+			} catch (Exception ex) {
+				throw new RuntimeException(ex);
+			}
+		}
+
+		private void collectResults() throws Exception {
```

Comment:
what do you mean?

Reply:
`methodNode` instead of `mn`, `insnNodeA` instead of `instr1`, I don't know what `s` is and the name should reflect its purpose. `ci` is obviously a shorthand for something, use the full name. `sv` ->` sourceValue`.

Reply:
Those are straight from stitch.

Reply:
The method or the variable names?

Reply:
variable names

Reply:
Well, stitch is using bad variable names 😛